### ISSUE TITLE
`Cannot invoke (class=VcsMappingChangedListener, method=mappingChanged, topic=VcsRepositoryMappingListener)`

### ISSUE THREAD

**Issue by MartinX3** [Blarc/gitlab-template-lint-plugin#120]
After observing the bug (it happens often), I think most of the time it happens if I open a folder as a project in IntelliJ which was cloned before with `git clone`.

```java
java.lang.RuntimeException: Cannot invoke (class=VcsMappingChangedListener, method=mappingChanged, topic=VcsRepositoryMappingListener)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:657)
	at com.intellij.util.messages.impl.MessageBusImplKt.deliverMessage(MessageBusImpl.kt:415)
	at com.intellij.util.messages.impl.MessageBusImplKt.pumpWaiting(MessageBusImpl.kt:394)
	at com.intellij.util.messages.impl.MessageBusImplKt.access$pumpWaiting(MessageBusImpl.kt:1)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:454)
	at jdk.proxy2/jdk.proxy2.$Proxy148.mappingChanged(Unknown Source)
	at com.intellij.dvcs.repo.VcsRepositoryManager.checkAndUpdateRepositoriesCollection(VcsRepositoryManager.java:361)
	at com.intellij.dvcs.repo.VcsRepositoryManager.initManager(VcsRepositoryManager.java:130)
	at com.intellij.dvcs.repo.VcsRepositoryManager$MyStartupActivity.runActivity(VcsRepositoryManager.java:85)
	at com.intellij.openapi.vcs.impl.VcsInitialization.lambda$runActivities$3(VcsInitialization.java:167)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:249)
	at com.intellij.openapi.vcs.impl.VcsInitialization.runActivities(VcsInitialization.java:167)
	at com.intellij.openapi.vcs.impl.VcsInitialization.runInitStep(VcsInitialization.java:149)
	at com.intellij.openapi.vcs.impl.VcsInitialization.execute(VcsInitialization.java:120)
	at com.intellij.openapi.vcs.impl.VcsInitialization$1.run(VcsInitialization.java:72)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:429)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:480)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:252)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:252)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "com.github.blarc.gitlab.template.lint.plugin.settings.remote.Remote.getGitlabUrl()" because "it" is null
	at com.github.blarc.gitlab.template.lint.plugin.listeners.VcsMappingChangedListener.mappingChanged(VcsMappingChangedListener.kt:32)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeMethod(MessageBusImpl.kt:677)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:640)
	... 34 more
```

**Comment by Blarc** [comment]
Thanks for reporting, will check it out.

**Comment by Blarc** [comment]
A fix was released under version 1.12.0. You can wait for JetBrains to approve version 1.12.0 or download it manually from https://github.com/Blarc/gitlab-template-lint-plugin/releases/tag/v1.12.0 and installing it following these [steps](https://github.com/Blarc/gitlab-template-lint-plugin#installation-from-zip).

Thanks for reporting back. :raised_hands:

**Comment by MartinX3** [comment]
With `v1.12.0`:
```
java.lang.RuntimeException: Cannot invoke (class=VcsMappingChangedListener, method=mappingChanged, topic=VcsRepositoryMappingListener)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:657)
	at com.intellij.util.messages.impl.MessageBusImplKt.deliverMessage(MessageBusImpl.kt:415)
	at com.intellij.util.messages.impl.MessageBusImplKt.pumpWaiting(MessageBusImpl.kt:394)
	at com.intellij.util.messages.impl.MessageBusImplKt.access$pumpWaiting(MessageBusImpl.kt:1)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:454)
	at jdk.proxy2/jdk.proxy2.$Proxy154.mappingChanged(Unknown Source)
	at com.intellij.dvcs.repo.VcsRepositoryManager.checkAndUpdateRepositoriesCollection(VcsRepositoryManager.java:361)
	at com.intellij.dvcs.repo.VcsRepositoryManager.initManager(VcsRepositoryManager.java:130)
	at com.intellij.dvcs.repo.VcsRepositoryManager$MyStartupActivity.runActivity(VcsRepositoryManager.java:85)
	at com.intellij.openapi.vcs.impl.VcsInitialization.lambda$runActivities$3(VcsInitialization.java:167)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:249)
	at com.intellij.openapi.vcs.impl.VcsInitialization.runActivities(VcsInitialization.java:167)
	at com.intellij.openapi.vcs.impl.VcsInitialization.runInitStep(VcsInitialization.java:149)
	at com.intellij.openapi.vcs.impl.VcsInitialization.execute(VcsInitialization.java:120)
	at com.intellij.openapi.vcs.impl.VcsInitialization$1.run(VcsInitialization.java:72)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:429)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:480)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:252)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:252)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "com.github.blarc.gitlab.template.lint.plugin.settings.remote.Remote.getGitlabUrl()" because the return value of "java.util.Map$Entry.getValue()" is null
	at com.github.blarc.gitlab.template.lint.plugin.listeners.VcsMappingChangedListener.mappingChanged(VcsMappingChangedListener.kt:32)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeMethod(MessageBusImpl.kt:677)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:644)
	... 34 more
```

**Comment by Blarc** [comment]
@MartinX3 Could you please try uninstalling the plugin and reinstalling it? Hopefully this deletes the plugin's settings xml file and solves the problem?

**Comment by MartinX3** [comment]
Then i would need to load every project where this xml file is before installing.

Would it be enough to edit the xml file in each peoject where the error occurs?

Also would it be possible to add a default value in gitlab to use gitlab.org or somehow add a local linter?

And would it be maybe possible to show an error dialog in the IDE if there is an invalid value in the options?

**Comment by Blarc** [comment]
Need to analyse a bit more why this happens and handle it better.

**Comment by MartinX3** [comment]
Thank you very much. :)

**Comment by Blarc** [comment]
A fix was released under version 1.13.0. You can wait for JetBrains to approve version 1.13.0 or download it manually from https://github.com/Blarc/gitlab-template-lint-plugin/releases/tag/v1.13.0 and installing it following these [steps](https://github.com/Blarc/gitlab-template-lint-plugin#installation-from-zip).

Thanks for reporting back. :raised_hands:
